sklearn/ensemble/forest.py

```diff
@@ -242,6 +243,11 @@ def fit(self, X, y, sample_weight=None):
         -------
         self : object
         """
+        # Validate hyperparameters
+        if self.n_estimators == 10:
+            warnings.warn("'n_estimators' default value will change to 100 in version 0.20", DeprecationWarning)
```

Comment:
We don't want to break the user code.
Currently, with your PR, a user using the default value will witness a change in its code, in particular increasing its fit time.

We probably want to just warn about *future* change:
- we should not change the value to 100
- the DeprecationWarning should be a FutureWarning
- the warning needs to say "will be changed in 0.22", instead of "has changed in 0.20"
- the warning could also be more pedagogical, explaining this value needs to be changed manually, as suggested in the issue https://github.com/scikit-learn/scikit-learn/issues/11128#issuecomment-392221494